p2p/switch.go

```diff
@@ -398,16 +399,22 @@ func (sw *Switch) IsDialing(addr *NetAddress) bool {
 // trying to send for defaultSendTimeoutSeconds. Returns a channel
 // which receives success values for each attempted send (false if times out).
 // NOTE: Broadcast uses goroutines, so order of broadcast may not be preserved.
-// TODO: Something more intelligent.
 func (sw *Switch) Broadcast(chID byte, msg interface{}) chan bool {
```

Comment:
Regarding issue #805: What if we kept the implementation similar to this, yet each broadcast was queued, so that broadcast X must finish before broadcast Y? The individual messages could be sent via go-routines, but the order of broadcasts would be preserved. Thoughts?

Reply:
Even better to have a queue for each peer. That way we won't be waiting for some peer!